packages/@aws-cdk/integ-runner/lib/runner/integ-test-runner.ts

```diff
@@ -238,26 +245,79 @@ export class IntegTestRunner extends IntegRunner {
           lookups: this.expectedTestSuite?.enableLookups,
         });
       }
+      // now deploy the "actual" test. If there are any assertions
+      // deploy the assertion stack as well
       this.cdk.deploy({
         ...deployArgs,
         lookups: this.actualTestSuite.enableLookups,
-        stacks: actualTestCase.stacks,
+        stacks: [
+          ...actualTestCase.stacks,
+          ...actualTestCase.assertionStack ? [actualTestCase.assertionStack] : [],
+        ],
+        rollback: false,
         output: this.cdkOutDir,
         ...actualTestCase?.cdkCommandOptions?.deploy?.args,
+        ...actualTestCase.assertionStack ? { outputsFile: path.join(this.cdkOutDir, 'assertion-results.json') } : undefined,
         context: this.getContext(actualTestCase?.cdkCommandOptions?.deploy?.args?.context),
         app: this.hasTmpActualSnapshot() ? this.cdkOutDir : this.cdkApp,
       });
+
       if (actualTestCase.hooks?.postDeploy) {
         exec([chain(actualTestCase.hooks?.postDeploy)], {
           cwd: path.dirname(this.snapshotDir),
         });
       }
+
+      if (actualTestCase.assertionStack) {
+        return this.processAssertionResults(
+          path.join(this.directory, this.cdkOutDir, 'assertion-results.json'),
+          actualTestCase.assertionStack,
+        );
+      }
     } catch (e) {
       this.parseError(e,
         actualTestCase.cdkCommandOptions?.deploy?.expectError ?? false,
         actualTestCase.cdkCommandOptions?.deploy?.expectedMessage,
       );
     }
+    return;
+  }
+
+  /**
+   * Process the outputsFile which contains the assertions results as stack
+   * outputs
+   */
+  private processAssertionResults(file: string, assertionStackId: string): AssertionResults | undefined {
```

Comment:
Is there a need for this to return `undefined`? Can we not just return an empty object?

Motivation: I like avoiding `if`s and special cases as much as I can.

Reply:
I need to check if the results have any assertion failures so I can fail the test and report on the failures. I would either need to check for `undefined` or an empty object and `undefined` seemed better. I can be persuaded otherwise if you think it is not though.